src/AggregateFunctions/AggregateFunctionGroupFormat.cpp

```diff
@@ -0,0 +1,238 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+
+#include <Columns/ColumnString.h>
+#include <Core/Block.h>
+#include <Core/ProtocolDefines.h>
+#include <Core/Settings.h>
+#include <DataTypes/DataTypeString.h>
+#include <Formats/FormatFactory.h>
+#include <Formats/NativeReader.h>
+#include <Formats/NativeWriter.h>
+#include <IO/VarInt.h>
+#include <IO/WriteBufferFromString.h>
+#include <Interpreters/Context.h>
+#include <Processors/Formats/IOutputFormat.h>
+
+#include <Common/assert_cast.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+extern const int LOGICAL_ERROR;
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+
+struct GroupFormatData
+{
+    MutableColumns columns;
+};
+
+class AggregateFunctionGroupFormat final : public IAggregateFunctionDataHelper<GroupFormatData, AggregateFunctionGroupFormat>
+{
+public:
+    AggregateFunctionGroupFormat(
+        const DataTypes & argument_types_,
+        const Array & parameters_,
+        String format_name_,
+        FormatSettings format_settings_,
+        ContextPtr context_)
+        : IAggregateFunctionDataHelper<GroupFormatData, AggregateFunctionGroupFormat>(
+              argument_types_, parameters_, std::make_shared<DataTypeString>())
+        , format_name(std::move(format_name_))
+        , format_settings(std::move(format_settings_))
+        , context(std::move(context_))
+    {
+        size_t num_columns = argument_types.size();
+        for (size_t i = 0; i < num_columns; ++i)
+        {
+            String name = "c" + std::to_string(i + 1);
+            header.insert({argument_types[i], name});
+        }
+    }
+
+    String getName() const override { return "groupFormat"; }
+
+    bool allocatesMemoryInArena() const override { return false; }
+
+    void create(AggregateDataPtr __restrict place) const override /// NOLINT
+    {
+        new (place) GroupFormatData;
+        auto & data = this->data(place);
+        data.columns.reserve(argument_types.size());
+        for (const auto & type : argument_types)
+            data.columns.emplace_back(type->createColumn());
+    }
+
+    void add(AggregateDataPtr __restrict place, const IColumn ** columns, size_t row_num, Arena *) const override
```

Comment:
I think the batch methods should also be implemented, as they will be more optimal.